static/src/js/components/GranuleResults/GranuleResultsItem.js

```diff
@@ -2,32 +2,179 @@
+                  {dataLink.title}
+                </Dropdown.Item>
+              )
+            })
+          }
+        </Dropdown.Menu>
+      </Dropdown>
+    )
+  }
+  return (
+    <a
+      className="button granule-results-item__button"
+      href={dataLinks[0].href}
+      rel="noopener noreferrer"
+      title="Download single granule data"
+      target="_blank"
+    >
+      <i className="fa fa-download" />
+    </a>
+  )
+}
+
+DataLinksButton.propTypes = {
+  dataLinks: PropTypes.arrayOf(PropTypes.shape({})).isRequired
+}
+
 /**
  * Renders GranuleResultsItem.
  * @param {object} props - The props passed into the component.
  * @param {object} props.granule - Granule passed from redux store.
  */
-const GranuleResultsItem = ({ granule, isLast, waypointEnter }) => {
-  const title = granule.producer_granule_id ? granule.producer_granule_id : granule.title
-  const temporal = granule.formatted_temporal
+const GranuleResultsItem = ({
+  collectionId,
+  granule,
+  isLast,
+  location,
+  waypointEnter,
+  onExcludeGranule,
+  onFocusedGranuleChange
+}) => {
+  const handleRemoveClick = () => {
+    const {
+      id,
+      is_cwic: isCwic
+    } = granule
+
+    let hashedId
+    if (isCwic) hashedId = murmurhash3(id).toString()
+    onExcludeGranule({
+      collectionId,
+      granuleId: hashedId || id
+    })
+  }
+
+  const handleClickGranuleDetails = (granuleId) => {
+    onFocusedGranuleChange(granuleId)
+  }
+
+  const {
+    browse_flag: browseFlag,
+    formatted_temporal: formattedTemporal,
+    id,
+    links,
+    online_access_flag: onlineAccessFlag,
+    producer_granule_id: producerGranuleId,
+    thumbnail: granuleThumbnail,
+    title: granuleTitle
+  } = granule
+
+  const title = producerGranuleId || granuleTitle
+  const temporal = formattedTemporal
   const timeStart = temporal[0]
   const timeEnd = temporal[1]
-  const thumbnail = granule.browse_flag ? granule.thumbnail : false
+  const thumbnail = browseFlag ? granuleThumbnail : false
+
+  const createDataLinks = () => {
+    // All 'http' data links that are not inherited
+    const httpDataLinks = links.filter((link) => {
+      const {
+        href,
+        inherited = false,
+        rel
+      } = link
+
+      return href.indexOf('http') !== -1 && rel.indexOf('/data#') !== -1 && inherited !== true
+    })
+
+    // Strip filenames from httpDataLinks
+    const filenames = httpDataLinks.map(link => link.href.substr(link.href.lastIndexOf('/') + 1).replace('.html', ''))
+
+    // Find any 'ftp' data links that are not interited that have filenames not already included with 'http' links
+    const ftpLinks = links.filter((link) => {
+      const {
+        href,
+        inherited = false,
+        rel
+      } = link
+
+      const filename = href.substr(href.lastIndexOf('/') + 1)
+
+      return href.indexOf('ftp://') !== -1 && rel.indexOf('/data#') !== -1 && inherited !== true && filenames.indexOf(filename) === -1
```

Comment:
This is a lot of code for one line -- can this be a util that we can add comments to for readability?

Reply:
@macrouch ill let you figure this one out :)

Reply:
I understand this is likely a port, but we should actively improve these types of things as we port them over.